ansible/playbooks/roles/upgrade/tasks/kubernetes.yml

```diff
@@ -1,25 +1,65 @@
 ---
+# During HA control plane upgrade server address in kubeconfig is switched to local for
+#   * compatibility between client and server versions
+#   * identifying correct server version
+
+- name: k8s/master | Switch apiserver address to local
+  include_tasks: kubernetes/utils/set-local-apiserver.yml # sets kubectl_context_cluster
+  when:
+    - groups.kubernetes_master | length > 1
+    - inventory_hostname in groups.kubernetes_master
+
 - name: Wait for kube-apiserver and get cluster version
-  delegate_to: "{{ groups.kubernetes_master[0] }}"
+  delegate_to: >-
+    {{ inventory_hostname if inventory_hostname in groups.kubernetes_master else
+      groups.kubernetes_master[0] }}
   block:
     - name: k8s | Include wait-for-kube-apiserver.yml
       import_tasks: kubernetes/utils/wait-for-kube-apiserver.yml
 
     - name: k8s | Include get-cluster-version.yml
```

Comment:
I think we can simplify this flow. To decide about skipping upgrade we need only two common things (doesn't matter if it is master or worker node):
1) flag file existence
2) kubelet version

This means that probably we do not have to get cluster version here but only later, when condition is met and upgrade is not skipped.

Tasks `Create K8s upgrade state file on master node` and `Create K8s upgrade state file on node` have the same content so we can have one common task even not using YAML anchors. We could set fact node_role { master | worker }, then have:
```
- name: k8s/{{ node_role  }} | Create K8s upgrade state file
   copy:
      dest: "{{ kubernetes.upgrade_state_file_path }}"
      content: Upgrade started
      mode: u=rw,g=r,o=
```
The same with removal tasks.

Reply:
As discussed with @to-bar and @seriva, we will not refactor that part for current release as it is more serious code interference with the upgrade of K8s.
Ticket for refactoring -> [link](https://app.zenhub.com/workspaces/epiphany-platform-5ef9d3c749afc20019782391/issues/epiphany-platform/epiphany/2844)